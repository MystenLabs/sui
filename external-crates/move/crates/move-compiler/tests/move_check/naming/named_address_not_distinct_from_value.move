// These checks straddle a few different passes but
// Named addresses are no longer distinct from their value
// This is due to the package system mostly maintaining the value

module A::M {
    const C: u64 = 0;
    struct S {}
    public fun s(): S { S{} }
}

module A::Ex0 {
    friend 0x41::M;
}

module A::Ex1 {
    use 0x41::M;
    public fun ex(): 0x41::M::S {
        0x41::M::C;
        0x41::M::s()
    }

    public fun ex2(): M::S {
        ex()
    }
}
