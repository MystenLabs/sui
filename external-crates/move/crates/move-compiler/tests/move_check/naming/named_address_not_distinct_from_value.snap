error[E04001]: restricted visibility
   ┌─ tests/move_check/naming/named_address_not_distinct_from_value.move:18:9
   │
 6 │     const C: u64 = 0;
   │           - Constants are internal to their module, and cannot can be accessed outside of their module
   ·
18 │         0x41::M::C;
   │         ^^^^^^^^^^ Invalid access of '0x41::M::C'

